clients/src/test/java/org/apache/kafka/test/MockConsumerInterceptor.java

```diff
@@ -52,6 +59,9 @@ public void configure(Map<String, ?> configs) {
 
     @Override
     public ConsumerRecords<String, String> onConsume(ConsumerRecords<String, String> records) {
+        if (CLUSTER_META.get() != null && CLUSTER_META.get().clusterId() != null && CLUSTER_META.get().clusterId().length() == 48)
+            IS_CLUSTER_ID_PRESENT_BEFORE_ON_CONSUME.set(true);
```

Comment:
That way logic like the length, what the field is, etc. can live outside this class that should be agnostic.